commons-statistics-descriptive/src/main/java/org/apache/commons/statistics/descriptive/SumOfSquaredDeviations.java

```diff
@@ -0,0 +1,121 @@
+ * {@link DoubleStatisticAccumulator#combine(DoubleStatistic) combine} method, it must be synchronized externally.
+ *
+ * <p>However, it is safe to use {@link java.util.function.DoubleConsumer#accept(double) accept} and {@link DoubleStatisticAccumulator#combine(DoubleStatistic) combine}
+ * as <code>accumulator</code> and <code>combiner</code> functions of
+ * {@link java.util.stream.Collector Collector} on a parallel stream,
+ * because the parallel implementation of {@link java.util.stream.Stream#collect Stream.collect()}
+ * provides the necessary partitioning, isolation, and merging of results for
+ * safe and efficient parallel execution.
+ */
+class SumOfSquaredDeviations extends FirstMoment implements DoubleStatistic {
+    /** Sum of squared deviations of the values that have been added. */
+    private double squaredDevSum;
+
+    /**
+     * Create a SumOfSquaredDeviations instance.
+     */
+    SumOfSquaredDeviations() {
+        // No-op
+    }
+
+    /**
+     * Create a SumOfSquaredDeviations instance with the given sum of
+     * squared deviations and a FirstMoment instance.
+     *
+     * @param squaredDevSum Sum of squared deviations.
+     * @param mean Mean of values.
+     * @param n Number of values.
+     * @param nonFiniteValue Sum of values.
+     */
+    SumOfSquaredDeviations(double squaredDevSum, double mean, long n, double nonFiniteValue) {
+        super(mean, n, nonFiniteValue);
+        this.squaredDevSum = squaredDevSum;
+    }
+
+    /**
+     * Updates the state of the statistic to reflect the addition of {@code value}.
+     * @param value Value.
+     */
+    @Override
+    public void accept(double value) {
+        super.accept(value);
+        squaredDevSum += (n - 1) * dev * nDev;
+    }
+
+    /**
+     * Gets the sum of squared deviations of all input values.
+     *
+     * @return {@code SumOfSquaredDeviations} of all values seen so far.
+     */
+    @Override
+    public double getAsDouble() {
+        return Double.isFinite(super.getAsDouble()) ? squaredDevSum : Double.NaN;
+    }
+
+    /**
+     * Gets a new FirstMoment instance with all of its parameters copied from the current instance.
+     * @return The FirstMoment instance.
+     */
+    FirstMoment getFirstMoment() {
+        return new FirstMoment(m1, n, super.getNonFiniteValue(), dev, nDev);
+    }
+
+    /**
+     * Combines the state of another {@code SumOfSquaredDeviations} into this one.
+     *
+     * @param other Another {@code SumOfSquaredDeviations} to be combined.
+     * @return {@code this} instance after combining {@code other}.
+     */
+    public SumOfSquaredDeviations combine(SumOfSquaredDeviations other) {
+        final long oldN = n;
+        final long otherN = other.n;
+        if (oldN == 0) {
+            squaredDevSum = other.squaredDevSum;
+        } else if (otherN != 0) {
+            final double diffOfMean = other.getFirstMoment().getAsDouble() - m1;
```

Comment:
Avoid creating a copy instance of the FirstMoment. Instead drop the implementation of the statistic interfaces (these are internal classes so do not require them):
```java
// In FirstMoment:
    double getFirstMoment() {
        if (Double.isFinite(m1)) {
            return n == 0 ? Double.NaN : m1;
        }
        // A non-finite value must have been encountered, return nonFiniteValue which represents m1.
        return nonFiniteValue;
    }

// In SumOfSquaredDeviations:
    final double diffOfMean = other.getFirstMoment() - m1
```